java/src/main/java/com/logicalclocks/hsfs/engine/FeatureGroupEngine.java

```diff
@@ -185,9 +186,20 @@ public void updateStatisticsConfig(FeatureGroup featureGroup) throws FeatureStor
     featureGroup.setHistograms(apiFG.getHistograms());
   }
 
-  public FeatureGroupCommit[] commitDetails(FeatureGroup featureGroup, Integer limit)
+  public Map<String, Map<String,String>> commitDetails(FeatureGroup featureGroup, Integer limit)
       throws IOException, FeatureStoreException {
-    return featureGroupApi.commitDetails(featureGroup, limit);
+    List<FeatureGroupCommit> featureGroupCommits = featureGroupApi.commitDetails(featureGroup, limit);
```

Comment:
If there are no commits, this is going to be null and the next part of the code will trigger a `NullPointerException`. Check for it.